proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1790 @@
+      throw an error.
+
+    * Set `alpha` to the result of [scaling] `alpha` by `$alpha` with `max` 1,
+      treating any `none` keyword as a value of `0`.
+
+* Let `channel-args` be the remaining keyword arguments in `$args`, not
+  including `$space` or `$alpha` arguments.
+
+* Let `channels` be a list of the `color`'s channels.
+
+* For each keyword `scale` in `channel-args`:
+
+  * If `scale` is not the name of a bounded or percentage-mapped channel in
+    `channels`, throw an error.
+
+  * Set the corresponding `channel` in `channels` to the result of [scaling]
+    `channel` by `scale` with a `max` defined by the channel boundary.
+
+* Let `normal` be the result of [normalizing] `channels`.
+
+* Return a color in color space `space`, with `normal` channels, and an alpha
+  of `alpha`.
+
+### `color.complement()`
+
+```
+complement($color, $space: null)
+```
+
+This function is also available as a global function named `complement()`.
+
+* If `$color` is not a color, throw an error.
+
+* If `$space` is null:
+
+  * If `$color` is a legacy color, let `space` be `hsl`.
+
+  * Otherwise, throw an error.
+
+* Otherwise:
+
+  * If `$space` is not a [known color space] with a polar-angle hue channel,
+    throw an error.
+
+    > This currently allows `hsl`, `hwb`, `lch`, and `oklch`. We may decide to
+    > provide additional options in the future.
+
+  * Let `space` be the value of `$space`.
+
+* Return the result of calling
+  `color.adjust($color, $hue: 180deg, $space: space)`.
+
+### `color.invert()`
+
+```
+invert($color, $space: null)
+```
+
+This function is also available as a global function named `invert()`.
+
+* If `$color` is not a color, throw an error.
+
+* If `$space` is null:
+
+  * If `$color` is a legacy color, let `space` be `rgb`, and let `mix-space`
+    be null.
+
+    > This allows us to also enforce legacy behavior in the final weighted mix.
+
+  * Otherwise, throw an error.
+
+* Otherwise:
+
+  * If `$space` is not a [known color space], throw an error.
+
+  * Let `space` be `$space`, and let `mix-space` be `$space`.
+
+* Let `color` be the result of [converting] and [gamut mapping] `$color`
+  into the color space `space`.
+
+* If `space == hwb`:
+
+  * Let `hue`, `whiteness`, and `blackness` be the three elements of `color`'s
+    channels.
+
+  * Let `hue-out` be the result of `(hue + 180deg) % 360deg`.
+
+  * Let `gray` be the result of `whiteness + blackness`.
+
+  * Let `white` be `gray - whiteness`, and let `black` be `gray - blackness`.
+
+  * Let `invert` be the result of calling
+    `color.change(color, $hue: hue-out, $whiteness: white, $blackness: black)`.
+
+* Otherwise:
+
+  * Let `invert` be the value of `color`.
+
+  * For each `channel` element in `color`'s channels:
+
+    * If `channel` represents a polar-angle hue:
+
+      * Let `new` be `(channel + 180deg) % 360deg`.
+
+    * Otherwise, if `channel` represents either chroma or saturation:
```

Comment:
"represents" is somewhat underspecified here. Maybe just say "if `channel`'s name is either `chroma` or `saturation`"